modules/gapi/include/opencv2/gapi/own/cvdefs.hpp

```diff
@@ -20,19 +20,20 @@ typedef          char schar;
 
 typedef unsigned short ushort;
 
+#define CV_USRTYPE1 (void)"CV_USRTYPE1 support has been dropped in OpenCV 4.0"
+
 #define CV_CN_MAX     512
 #define CV_CN_SHIFT   3
 #define CV_DEPTH_MAX  (1 << CV_CN_SHIFT)
 
-
 #define CV_8U   0
 #define CV_8S   1
 #define CV_16U  2
 #define CV_16S  3
 #define CV_32S  4
 #define CV_32F  5
 #define CV_64F  6
-#define CV_USRTYPE1 7
+#define CV_16F  7
```

Comment:
I'm not sure about it: because user sometimes may be wondered: why they custom type doesn't work as expected

Also, i think `#define CV_USRTYPE1 (void)"CV_USRTYPE1 support has been dropped in OpenCV 4.0"` will trigger compilation error in expanded context

Reply:
It is from here: https://github.com/opencv/opencv/blame/4.0.0/modules/core/include/opencv2/core/hal/interface.h#L67

Reply:
@alalek so is it OK for us to just copy `interface.h` (as it is done here now) ?

Reply:
may be describe about this custom type support drop in modules/gapi/doc/10-hld-overview.md ?

if such behavior is expected then LGTM